src/octoprint/static/js/app/helpers.js

```diff
@@ -1616,3 +1620,78 @@ var deepMerge = function (target, source) {
 
     return target;
 };
+
+var isWebRTCAvailable = function () {
```

Comment:
From my Googling, there didn't seem to be a canonical way of checking for WebRTC compatibility. The most fully-featured option seems to be [DetectRTC.js](https://github.com/muaz-khan/DetectRTC) but that sure seems like a heavy dependency to add for a simple check. I also saw articles like [this](https://blog.ivrpowers.com/post/coding/browser-webrtc/) but they seemed a little sketchy (ie does the presence of `navigator.getUserMedia` definitively mean WebRTC is supported and vice versa?).

So I opted for the simplest solution -- just check if RTCPeerConnection and all the methods we're going to call exist.

Related: There's also [adapter.js](https://github.com/webrtcHacks/adapter) which attempts to patch naming disparities in the WebRTC implementations and is recommended in the [Mozilla docs](https://developer.mozilla.org/en-US/docs/Web/API/WebRTC_API). If we start seeing wider WebRTC adoption in OctoPrint, then it might be worth bringing that dependency in to support a few more browsers.

Reply:
I would have gone with that solution as well given the situation 👍